test/kubernetes_test.go

```diff
@@ -12,63 +11,158 @@ import (
 	"github.com/miekg/dns"
 )
 
-// Test data for A records
-var testdataLookupA = []struct {
-	Query            string
-	TotalAnswerCount int
-	ARecordCount     int
-}{
-	// Matching queries
-	{"mynginx.demo.svc.coredns.local.", 1, 1}, // One A record, should exist
-
-	// Failure queries
-	{"mynginx.test.svc.coredns.local.", 0, 0},                     // One A record, is not exposed
-	{"someservicethatdoesnotexist.demo.svc.coredns.local.", 0, 0}, // Record does not exist
-
-	// Namespace wildcards
-	{"mynginx.*.svc.coredns.local.", 1, 1},                       // One A record, via wildcard namespace
-	{"mynginx.any.svc.coredns.local.", 1, 1},                     // One A record, via wildcard namespace
-	{"someservicethatdoesnotexist.*.svc.coredns.local.", 0, 0},   // Record does not exist with wildcard for namespace
-	{"someservicethatdoesnotexist.any.svc.coredns.local.", 0, 0}, // Record does not exist with wildcard for namespace
-	{"*.demo.svc.coredns.local.", 2, 2},                          // Two A records, via wildcard
-	{"any.demo.svc.coredns.local.", 2, 2},                        // Two A records, via wildcard
-	{"*.test.svc.coredns.local.", 0, 0},                          // Two A record, via wildcard that is not exposed
-	{"any.test.svc.coredns.local.", 0, 0},                        // Two A record, via wildcard that is not exposed
-	{"*.*.svc.coredns.local.", 2, 2},                             // Two A records, via namespace and service wildcard
+func makeRR(s string) dns.RR {
```

Comment:
test.A and test.MX our your friends here - at least they are used in other tests.